src/silo/storage/aa_store.cpp

```diff
@@ -13,22 +13,49 @@
 #include "silo/common/zstdfasta_reader.h"
 #include "silo/preprocessing/preprocessing_exception.h"
 
+silo::AAPosition::AAPosition(AA_SYMBOL symbol) {
+   symbol_whose_bitmap_is_flipped = symbol;
+}
+
+silo::AAPosition::AAPosition(std::optional<AA_SYMBOL> symbol) {
+   symbol_whose_bitmap_is_flipped = symbol;
+}
+
 void silo::AAPosition::flipMostNumerousBitmap(uint32_t sequence_count) {
+   std::optional<AA_SYMBOL> flipped_bitmap_before = symbol_whose_bitmap_is_flipped;
    std::optional<AA_SYMBOL> max_symbol = std::nullopt;
    uint32_t max_count = 0;
 
    for (const auto& symbol : AA_SYMBOLS) {
       roaring::Roaring bitmap = bitmaps.at(symbol);
       bitmap.runOptimize();
-      const uint32_t count = bitmap.cardinality();
+      bitmap.shrinkToFit();
+      const uint32_t count = flipped_bitmap_before == symbol ? sequence_count - bitmap.cardinality()
+                                                             : bitmap.cardinality();
       if (count > max_count) {
          max_symbol = symbol;
          max_count = count;
       }
    }
-   if (max_symbol.has_value()) {
+   if (max_symbol != flipped_bitmap_before) {
```

Comment:
Can we find a better name for `flipped_bitmap_before` because I find this comparison strange? Maybe something like `symbol_before_flipping_bitmaps`, or is this the wrong meaning?